### ISSUE TITLE
Moving jakartablogs.ee to jakarta.ee

### ISSUE THREAD

**Issue by karenmcnaughton** [jakartaee/jakarta.ee#1037]
For the website redesign of Jakarta EE, we want to create a blogging platform that will allow users to log in and publish blog posts. The platform must be hosted on blogs.jakarta.ee and ideally allow for moderation.
Timing of this is to correspond with the launch of the revised jakarta.ee website.

**Comment by aeiras** [comment]
Hola Karen, 

I am so happy to see this issue arrive. 
In MicroProfile for +4years, we have used WordPress to enable Content contributors to add content. 
We have in place the MicroProfile Content guidelines as a call to action in the MP website under [Blog tab ](https://microprofile.io/blog/)that brings anyone interested to the HOW to. 

I recommend Jakarta EE Content call to action uses a similar review & process. 
It has proven effortless and efficient for our community.

**Comment by ivargrimstad** [comment]
It will need to continue to be an aggregator such as https://jakartablogs.ee where bloggers can "syndicate" feeds from their blogs and not be required to copy/paste/format content somewhere else than where they usually produce the content.

**Comment by chrisguindon** [comment]
> For the website redesign of Jakarta EE, we want to create a blogging platform that will allow users to log in and publish blog posts. The platform must be hosted on blogs.jakarta.ee and ideally allow for moderation.
> Timing of this is to correspond with the launch of the revised jakarta.ee website.

To summarize, my understanding is that our requirements are the following:

- Create a blog section for jakarta.ee
- allow users to login and submit content
- Allow for moderation
- host the platform on blogs.jakarta.ee

Before we start, I have a few more questions:

**Do we need to create a separate site for this?** 

Hugo can be used for blogging and I would prefer that we find a solution that leverages our current infrastructure to reduce the maintenance burden for the Foundation. 

**Why do we have a requirement to host this under blogs.jakarta.ee?**

Is this a requirement or simply a preference over jakarta.ee/blogs?

I don't think there is a wrong answer to this but I would recommend reading the following blog post before making a decision: 
https://blog.cloudflare.com/subdomains-vs-subdirectories-best-practices-workers-part-1/

**Comment by aeiras** [comment]
Feliz 2021 @chrisguindon, 

Great follow up!

Ideally, the current blog website is deprecated. You are correct about nonsense infra upkeep. 

Brainstorming a few features:
- Blog as a standalone tab in the Jakarta EE Website
- Readers and contributors find how to contribute content directly there with a call 2 action bottom to guidelines that brings them to the Jkta  Repo
- Original content could be awesome, but that is something that could be added at a later time. Let's start small. 
- We want Volunteers from the community. The blog is not the responsibility of the Mktg cmtee. The community must own the responsibility to review & help publish blogs. When I think about the Blog's review, I think about using PRs to publish the content.  A contributor submits a PR, and PR is reviewed and merged, and the blog goes live. Done! can Hugo do that? If so, amazing!

In anything we do, let's make sure that it is always scalable, that you can use it for other WG projects, and that others within those communities can own the process without blockage from INFRA. That includes anyone who is a committer under the Jakarta EE project. 
- 

When rebuilding anything, enabling healthy contributions means to cater to the ones who will provide the content, in this case. 
 If we cannot provide solutions that make their lives easier, let's step back & think about other ways to achieve a long term outcome that takes into feedback and can be modified without barriers.

**Comment by chrisguindon** [comment]
Feliz 2021 @aeiras,

> Ideally, the current blog website is deprecated. You are correct about nonsense infra upkeep.

Unfortunately, I think we will need to keep jakartablogs.ee. As mention by Ivar, I think we still need a blog aggregator for authors who wishes to write content on their own website/platform.

Brainstorming a few features:

- Blog as a standalone tab in the Jakarta EE Website

What do you mean by a standalone tab? A menu item?

- Readers and contributors find how to contribute content directly there with a call 2 action bottom to guidelines that brings them to the Jkta Repo

+1 

- Original content could be awesome, but that is something that could be added at a later time. Let's start small.

What do you mean by original content could be awesome?

- We want Volunteers from the community. The blog is not the responsibility of the Mktg cmtee. The community must own the responsibility to review & help publish blogs. When I think about the Blog's review, I think about using PRs to publish the content. A contributor submits a PR, and PR is reviewed and merged, and the blog goes live. Done! can Hugo do that? If so, amazing!

Yes, we can definitely do that with Hugo. This is how most of the website is currently being updated. However, we will need to write some instructions.

**Comment by aeiras** [comment]
Hey Chris, 

- _What do you mean by a standalone tab? A menu item?_ 
Using [MP website](https://microprofile.io)- the blog is currently hidden in Resources. Every time without a fail, when I ask/suggest a Jakartee to submit content to Jakarta EE Blogs, none can find how to use the website. It is so frustrating that technology will choose to have such a hard way to enable contributors to help. We must do better. 

In MP we have many volunteers approving the blogs, it is amazing. Jakarta will get there as it continues to work via open channels and avoids vacuum nonsense. The future is exciting! 
I look forward to helping in any way I can push this improvement. Plus working with you and your team rocks! Thank you Chris!

**Comment by chrisguindon** [comment]
Is this still needed now that we have a new /news section on the website? https://jakarta.ee/news/

**Comment by aeiras** [comment]
@chrisguindon, 

I don't think we need this anymore.  Blogs is link to Website, yet it is hidden. My only comment here is to make it into a new tab that stands alone in the website, not hidden under Resources. 
The announcement section now allows for original content to live in the website, so in my view we have fixed 1 core feature that was not there before. 

Thanks for remembering this ticket.

**Comment by ivargrimstad** [comment]
I still think that `https://blogs.jakarta.ee` would be a better location for the blogs than `https://jakartablogs.ee`. And the blogs should be accessible from the top-level menu.

**Comment by aeiras** [comment]
+1 Ivar. Currently the blogs is its own website which I continue to forget. 

The goal is to get viewers to stay in Jakarta EE, not diverted to another place. 
In anything that leads to how we communicate project stuff, we must always fix whatever doesn't follow that norm.

**Comment by chrisguindon** [comment]
> I still think that `https://blogs.jakarta.ee` would be a better location for the blogs than `https://jakartablogs.ee`. And the blogs should be accessible from the top-level menu.

Blogs will become a top-level link with the new re-design:
https://projects.invisionapp.com/share/AX10M9QL74RN#/screens

https://jakartablogs.ee is on a separate domain for security reasons. The big concern is that an author's website can be hacked or they might forget to renew and someone else might pick it up. I am not making this up, this actually occurred a few times with planeteclipse.org.

A malicious user takes over and injects malicious code in a blog post. (Usually, they simply inject spam but they could do worst).

We keep it on a separate domain to protect jakarta.ee. We could argue that we don't do much with cookies on jakarta.ee and there is not much for them to steal.

However, this move could limit what we can do in the future with jakarta.ee since I wouldn't feel safe allowING users to login to jakarta.ee with that in place.

This is a similar approach that GitHub has with Github pages or Netlify.

The main website for Netlify is https://www.netlify.com/ but all apps are hosted under a separate domain (.netlify.app) such as jakartaee.netlify.app

Apps no longer have access to cookies under netlify.com.

**Comment by chrisguindon** [comment]
If we really want to do this, we will need to implement some very strict filtering rules on jakartablogs.ee to make sure that only safe HTML elements are being used on the site. 

We will need to make sure that anything else gets stripped out. If we do so, I will be ok with setting up a reverse proxy rule to host blogs under blogs.jakarta.ee.

**Comment by cesarhernandezgt** [comment]
Hi all, 

I would like to provide my view and hopefully facilitate pointers that help to understand better the scope of the current Github issue.

@chrisguindon @ivargrimstad 
At the beginning of 2021, in JakartaEE Marketing Committee, we receive the "Jakarta EE SEO & PPC Overview" report. You can sync with @karenmcnaughton to obtain a better view of the recommendations since this github issue is one of the recommended actions in the report.


@ivargrimstad 
> It will need to continue to be an aggregator such as https://jakartablogs.ee where bloggers can "syndicate" feeds from their blogs and not be required to copy/paste/format content somewhere else than where they usually produce the content.


This GitHub issue title refers to the migration of jakartablogs.ee to jakarta.ee. 
This Github issue description extends a bit the scope, which I'm totally +1 because states "blogging platform that will allow users to log in and publish blog posts.  .... and ideally, allow for moderation." Base on the 2021's SEO feedback, redirects to syndicate blogs feeds is fine but we are lacking content hosted in the Jakarta.ee website. The option to syndicate feed can be migrated but we also need a way to allow content to be generated, moderated, and hosted in jakarata.ee website. You can check page #3 of the "Jakarta EE SEO & PPC Overview" report.

> I still think that https://blogs.jakarta.ee would be a better location for the blogs than https://jakartablogs.ee. And the blogs should be accessible from the top-level menu.

+1

@chrisguindon 

> Do we need to create a separate site for this?
> Hugo can be used for blogging and I would prefer that we find a solution that leverages our current infrastructure to reduce the maintenance burden for the Foundation.

That is a good question, without being an expert in Hugo I also agree that current infrastructure can be used without adding more complexity or technical debt.

> Why do we have a requirement to host this under blogs.jakarta.ee?
> Is this a requirement or simply a preference over jakarta.ee/blogs?

Page #3 of the "Jakarta EE SEO & PPC Overview" report explicitly indicates "blogs.jakarta.ee". My suggestion would be to circle back your observation to @karenmcnaughton or @TanjaObradovic so you can have a confirmation back from SEO team.

> Is this still needed now that we have a new /news section on the website? https://jakarta.ee/news/

The short answer would be, yes. Current https://jakarta.ee/news/ represent official news and announcement from the Jakarta EE Working group. 
The scope of this Github issue is related to the migration of jakartablogs.ee into Jakarta.ee along with the features of moderation and content hosting. Notice that blog posts can be contributions coming from committers, participant members, and the overall community to share written content.

> https://jakartablogs.ee is on a separate domain for security reasons. The big concern is that an author's website can be hacked or they might forget to renew and someone else might pick it up. I am not making this up, this actually occurred a few times with planeteclipse.org.

Thank you for the context, I was not aware of it. I don't have a straight proposal to fix this particular point. For a contributor who wants to write and post a new blog post under jakarta.ee I think event Github PR workflow suffice the requirement. 

But for crossposting moderation, I wonder if Hugo can handle some kind of lazy sync between another repository? 
I'm brainstorming in the following workflow:

1. I'm a new contributor who has blog and goest to blogs.jakarta.ee (or jakarta.ee/blogs) and fill in the form to submit my RSS 

2. A second repository will be actually in charge to perform the work that current jakartablogs.ee backend does, but nothing gets publicly available nor committed yet on jakarta.ee repository. 

3. The review can be done from the snapshot taken from step 2.

4. If the review is successful, the snapshot of the article will be migrated into blogs.jakarta.ee (or jakarta.ee/blogs).

Notice that with this workflow we would keep static content hosted on our side independently if at some point the original blog post get updated with malicious code. Notice also that I'm not thinking about the technical challenges yet, just the overall workflow of having the final content hosted under jakarta.ee. Yes, moderation of individual articles from already subscribed blog post will need it, as far as I know, were currently don't do that, but I think that would increase the overall security too.

**Comment by chrisguindon** [comment]
> @chrisguindon @ivargrimstad
> At the beginning of 2021, in JakartaEE Marketing Committee, we receive the "Jakarta EE SEO & PPC Overview" report. You can sync with @karenmcnaughton to obtain a better view of the recommendations since this github issue is one of the recommended actions in the report.

I have seen the report. We made some updates to our website to address some issues. I also reach out to the specification team to fix issues on their end: https://github.com/jakartaee/specifications/issues/318


> That is a good question, without being an expert in Hugo I also agree that current infrastructure can be used without adding more complexity or technical debt.

I would add that the work done for https://jakarta.ee/news/ could be re-purposed to create a new blog section if that is needed.

> Page #3 of the "Jakarta EE SEO & PPC Overview" report explicitly indicates "blogs.jakarta.ee". My suggestion would be to circle back your observation to @karenmcnaughton or @TanjaObradovic so you can have a confirmation back from SEO team.

I refer back to my previous comment for this:
https://github.com/jakartaee/jakarta.ee/issues/1037#issuecomment-833629475

It's possible but we need to make sure that content on blogs.jakarta.ee is secure. We will need to review the allowed HTML tags that we publish on that site. 

If we want to do this, we need the community to get involved in that project to make progress on this but it's possible: https://github.com/jakartaee/jakartablogs.ee

> But for crossposting moderation, I wonder if Hugo can handle some kind of lazy sync between another repository?
> I'm brainstorming in the following workflow:

Do you mean something similar to how we deploy the specification section on the website? It's a separate repo https://github.com/jakartaee/specifications and every night a corn job runs which pulls in the new content https://github.com/jakartaee/jakarta.ee/blob/src/build.sh

**Comment by cesarhernandezgt** [comment]
> I would add that the work done for https://jakarta.ee/news/ could be re-purposed to create a new blog section if that is needed.

+1 if we can reuse current logic and code behind /news/ for the creation of new blog posts hosted in the website itself. cc: @Dexmaster 


> If we want to do this, we need the community to get involved in that project to make progress on this but it's possible: https://github.com/jakartaee/jakartablogs.ee

Technically speaking, for me is a natural step choosing `jakarta.ee/blogs` it also integrates smoothly with the above idea of reusing /news logic and code to offer /blogs list and blog posts. What do you think @karenmcnaughton @TanjaObradovic?


> Do you mean something similar to how we deploy the specification section on the website?

Yes, that's right.

**Comment by Dexmaster** [comment]
> +1 if we can reuse current logic and code behind /news/ for the creation of new blog posts hosted in the website itself. cc: @Dexmaster

Yes, copying current logic from news to block shouldn't be hard or complicated. My only suggestion would be also renaming css classes so they will have distinct class naming, and in future their look could be separated if you chose to do so.

As for `jakarta.ee/blogs` vs `jakartablogs.ee`, I would suggest using `jakarta.ee/blogs`. Maintaining two different resources with different functionality is in my opinion counterproductive. In future if you will have comments or huge article load it might need separation. But as far as I've seen at the moment it just complicates structure, without providing any benefits. You can also do separation as a data-source for hugo (basically saving all blog posts in md/adoc format in structure you need and pulling changes on build or including repository as child folder), not new design or tools.

Whichever form you'll go I will suggest allowing only md/adoc + pdf and image files for blog articles (we have html disabled, so there shouldn't be any content issues). 
Plus you can add to cd/ci:
- cleaning image metadata + limiting image size on each PR merge (you already have optimize_news_images script which does all you need, just tweak it for blogs)
- markdown and adoc proofers/linters (here you will need to select tools you'll prefer, but as a bonus it will fix other pages typos/markdown issues, and mention to authors they have issues with PR before you even look at it)

This way you should have no issues with content itself.

**Comment by chrisguindon** [comment]
There wouldn't be any problems with hosting the articles under jakarta.ee/blogs if moderate the content before they are posted on the site!

**Comment by aeiras** [comment]
Hola Chris @chrisguindon, 

Crossposting via PR will continue to be the model for upkeep, correct?

How about hosting original content as we now do for the Announcement space? Not everyone has their own website or chooses to use GitHub or medium or similar to share stuff. Some prefer to write to hope that the content is kept under the project's blog as the sole bookkeeper.  

Thanks for the follow-up,

**Comment by thatsmekristy** [comment]
Looping in @shabnammayel

Hi @chrisguindon, 
Perhaps the decision has already been made but as the person who wrote the original recommendation to move the blog, using jakarta.ee/blogs would be even better than blogs.jakarta.ee for building domain authority. If you wanted to have a separate CMS for the blog, using the subdomain blogs.jakarta.ee would be the next best option.

Is there a scheduled release date for this? Would love to get this in place.

**Comment by chrisguindon** [comment]
We can definitely create a blog section on the Jakarta.ee website, that is not a problem. I agree that jakarta.ee/blogs would be best for this. 

If agreed upon, we can include this new section for phase 2 of the redesign in Q4 as this was the original intention for this issue.  It's currently not included in our plans since a blog section was not included in the mockup. We will need to address that depending on what we agree to do here. (We don't need a mock-up if we re-use work that was already contributed)

> Timing of this is to correspond with the launch of the revised jakarta.ee website.

For more information about that, you can check out: https://github.com/jakartaee/jakarta.ee/issues/1236 and https://redesign--jakartaee.netlify.app/.

Migrating jakartablogs.ee to jakarta.ee is possible but might not be required If we create a new blog section on jakarta.ee. We can make sure that posts are written there appear on jakartablogs.ee.

If we absolutely want to move the jakartablogs app under jakarta.ee, we can also do that but we need to reserve some time to update our template to make sure that it's not possible to inject malicious code. To speed this up, a contribution from the community would definitely help: https://github.com/jakartaee/jakartablogs.ee

I think we can solve this by simply switching from full articles to a list of articles in a teaser view mode. Similar to how we do blogs at EclipseFdn: https://blogs.eclipse.org/.

This is the main reason why the jakartablogs.ee website is not under jakarta.ee. The impact of running malicious code on jakartablogs.ee is lower than jakarta.ee even if we want to avoid both.

This app aggregate blogs post without any moderation. We've already seen in the past people picking up expiring domains to post spam on planeteclipse.org.

Jakartablogs.ee is a fork of planeteclipse.org.

Assuming, we decide that we want to do both. We will need to finalize the requirements for all of this so that I know exactly what is expected from us. If we do that in September, I will make it a priority for my team to solve both issues in Q4.

**Comment by chrisguindon** [comment]
A lot has been discussed on this issue. Let's take a step back and finalize the requirements:

When Karen initially created this issue, she did mention that she expected users to be able to login and use a CMS. 

Since then, we've been discussing solutions to include the blog section within Hugo. Similar to how the community implemented our WG News section: https://jakarta.ee/news/

Do we still agree this is the path forward?

**Comment by chrisguindon** [comment]
Here's what I think the requirements are so far:

- we need to allow staff and the community to submit blog posts under jakarta.ee/blogs via a PR
- we need to display blogs posts from our blog aggregator (jakartablogs.ee) on jakarta.ee and/or migrate jakartablogs.ee to blogs.jakarta.ee

Is this correct or am I missing anything?

**Comment by aeiras** [comment]
Dias Chris, 

On point 2, migration of jakartablogs to main website would be ideal as the readers will stay on the main website, less clicking, etc. 
If we could terminate the blog aggregator, that would be a win to bring everything relevant into 1 house. 

What are the issues with such a task?

**Comment by chrisguindon** [comment]
The blog aggregator is needed because we need a service responsible for aggregating blog posts from multiple sources. 

Jakartablogs.ee is more than a website. Within that service, there is a process that runs periodically to see if new posts were published.

With that said, we can leverage this service to display posts and/or a list of posts on jakarta.ee. For example, we could leverage the RSS feed generated by the jakartablogs.ee to display posts on jakarta.ee: https://jakartablogs.ee/rss20.xml

I think that would be my preferred solution since that would allow us to display posts on jakarta.ee without requiring an update to jakartablogs.ee. (We can remove HTML tags from the RSS feed before displaying the content on the page)

## How could this be represented on the website?

An idea that was suggested was to leverage the work that was done by the community for the WG news section:
https://jakarta.ee/news/

Keeping that layout in mind, I could see using the main content section (left side) to display blog posts submitted via a PR. 

This would allow us to leverage the sidebar to display "Jakartablogs.ee" posts in the same format as the News items:

![Screenshot 2021-09-13 134337](https://user-images.githubusercontent.com/893942/133131453-1059a782-e37b-4f88-8492-8e12d0ed5fe0.png)

**Comment by cesarhernandezgt** [comment]
Personal feedback:

> Looping in @shabnammayel
...using jakarta.ee/blogs would be even better than blogs.jakarta.ee for building domain authority. ...
> 

+1


> we need to allow staff and the community to submit blog posts under jakarta.ee/blogs via a PR

+1 and this requires minor time and effort if we reuse (copy) what we currently have for jakarta.ee/news 
cc @Dexmaster 

> we need to display blogs posts from our blog aggregator (jakartablogs.ee) on jakarta.ee and/or migrate jakartablogs.ee to blogs.jakarta.ee

+1

As pointed out by @chrisguindon, this has more than one path to follow. My point of view on this topic is that we can have a blog aggregator GitHub repo (jakartablog.ee repositocory for example) and automation will need to be implemented  (maybe Github actions or Eclipse Jenkins CI) so PR's containing the Title and Redirect link to the original third-party article can be created as an entry on jakartalblogs.ee/blogs. This will act as a moderation process that we currently don't have to make sure third-party articles are safely posted on  jakartalblogs.ee/blogs.

**Comment by chrisguindon** [comment]
We had a call today with the Jakarta.EE Marketing Committee to discuss the requirements for this issue. We did agree on the requirements that were listed in my previous comment: 

1. we need to allow staff and the community to submit blog posts under jakarta.ee/blogs via a PR
2. we need to display blogs posts from our blog aggregator (jakartablogs.ee) on jakarta.ee and/or migrate jakartablogs.ee to blogs.jakarta.ee

The priority in Q4 is requirement #1. With the work that we are currently doing with the Jakarta.ee website re-design, we will include a new blog section to allow staff and the community to publish content via a PR.

In parallel, we are going to work on a solution that will allow us to include blog posts from jakaratablogs.ee in the new blog section of jakarta.ee. 

The plan for the moment would be to modify the POC that @cesarhernandezgt created https://github.com/cesarhernandezgt/blog-buffet/. We need the GA to create a markdown file for each post instead of including all the links in 1 document.

The main concern that I have with this solution is "content duplication". We will need to set a unique filename that we will use as the unique identifier for each blog post to avoid duplication.

An idea would be to use the permalink value in the RSS feed(https://jakartablogs.ee/rss20.xml) as the name of the markdown file. Hopefully, this will allow the system to properly create new documents for new posts and update existing posts if a change was made by the author.

We don't plan on embedding the full content of these blog posts. Instead, we will simply redirect the user to the original article.

Please let me know if I am missing anything from our call!

**Comment by chrisguindon** [comment]
I am also working on a similar problem for the ecdtools.eclipse.org and I just made a proposal that I think might work for both WG websites: https://github.com/EclipseFdn/ecdtools.eclipse.org/issues/160

Before I start, I am going to remind everyone of the requirements:

- we need to allow staff and the community to submit blog posts under jakarta.ee/blogs via a PR
- we need to display blogs posts from our blog aggregator (jakartablogs.ee) on jakarta.ee and/or migrate jakartablogs.ee to blogs.jakarta.ee

---

For requirement 1, we plan on creating a new blog section similar to the news section. @ericpoirier Is planning to get started on that soon. This new blog section will have an RSS feed that we can use. Since the blog section does not exist yet in Jakarta.ee, I am going to share the one from https://ecdtools.eclipse.org/resources/blogs/index.xml as an example

For requirement 2: Jakartablogs also generates an XML feed for all the posts: https://jakartablogs.ee/rss20.xml

I am proposing that we combine both feeds before displaying the list of blogs posts on jakarta.ee/blogs.

This should allow us to display all the posts written on jakarta.ee with all the posts currently available on jakartablogs.ee. By doing so, I can make sure to remove any un-safe HTML tags from the jakarta.ee website.

I currently prefer this solution since it removes the extra maintenance that would be involved if a new PR is created each time a new post is added to jakartablogs.ee. 

What does everyone think about this idea?

**Comment by chrisguindon** [comment]
I made some progress on this for the ECDTOOLS.eclipse.org:

Preview URL: https://deploy-preview-193--eclipsefdn-ecdtools.netlify.app/resources/blogs/
Related issue: https://github.com/EclipseFdn/ecdtools.eclipse.org/issues/160#issuecomment-986969583

In this preview, we are fetching blog posts from 2 different sources (the website and planeteclipse.org (However, we only include posts with the EDCTOOLS tag from planeteclipse.org for obvious reasons)).

- https://planeteclipse.org/planet/ecdtools.xml
- https://ecdtools.eclipse.org/resources/blogs/index.xml

I still need to clean up the code but if there is no objection, we plan on implementing the same solution for jakarta.ee. The only difference is that we will be pulling in blogs posts from jakartablogs.org instead of planeteclipse.org.

I expect the list of posts to be very long but we can plan on adding pagination in the new year if that's an issue.

**Comment by chrisguindon** [comment]
This feature is now live for https://ecdtools.eclipse.org/resources/blogs/. It's also used on the front page for their Whats New? section. This new widget will analyze all feeds and display the latest post. 

In theory, we could look at doing the blog section to Jakarta.ee **What's new** section, if this is something we would like to do.

The posts are coming from https://ecdtools.eclipse.org/resources/blogs/index.xml and https://planeteclipse.org/planet/ecdtools.xml

Our next step is to implement this feature for Jakarta.ee with the new blog section that @ericpoirier is currently working on: https://github.com/jakartaee/jakarta.ee/pull/1329

**Comment by ericpoirier** [comment]
@chrisguindon @shabnammayel I created  the following staging page for the Blogs section which is fetching data from JarkartaBlogs as well as the internal Jakarta EE blog posts. Note that there are currently no internal blog posts so for now only JakartaBlogs posts are showing.

https://deploy-preview-1329--jakartaee.netlify.app/blogs/

Since there are no mockup for that page I decided to use the same blocks we are using throughout our website to separate the blogs on that page.

Let me know what you think and if we're missing something.

Thanks.

**Comment by shabnammayel** [comment]
Can we add the dates similar to this page?https://jakarta.ee/news/ and also add images to the blog similar to the news page?

**Comment by chrisguindon** [comment]
We will be able to use images for blog posts that will be written for the jakarta.ee website. 

For example, here's a post that Brian wrote for the ECDTOOLS website: 
https://ecdtools.eclipse.org/resources/blogs/2021-11-02-embedded/

**Comment by ericpoirier** [comment]
> Can we add the dates similar to this page?https://jakarta.ee/news/ and also add images to the blog similar to the news page?

@shabnammayel I think its a good idea to update the layout for the dates like the news page but at this point I think this is something we could improve in Q1. I think we should focus on merging what we have right now in order to accomplish our MBOs and add that as a todo next year.

I can open an issue to improve the layout of the blogs page so that it looks like the news page.

**Comment by chrisguindon** [comment]
I tend to agree with Eric. Our focus this quarter was finding a way to display blogs on jakartablogs.ee on jakarta.ee. However, there is still a lot of work to be done with blogs next quarter.

We need to define guidelines and process to enable staff and community members for submitting a blog post on jakarta.ee. At the same time, we could look at improving the list of blogs.

**Comment by ericpoirier** [comment]
Also to answer the image question. I think @shabnammayel means images on the blogs page and not in individual blog posts.

In order to add images in  the summary, we will also have to update the script to fetch them from the blog sources.

**Comment by chrisguindon** [comment]
We have more flexibility with posts written on the website versus those on jakartablogs.ee. For those posts, the best thing we could do is pick an image from the content: https://jakartablogs.ee/rss20.xml

However, with posts on jakarta.ee, we can create a page parameter that would allow us to associate an image to a blog post. 

We would probably need to update the XML template of the site to share that info with the script.

These are things we could look at doing in Q1 but I personally would not consider these details to be a show stopper from displaying jakartablogs.ee posts to jakarta.ee.

**Comment by shabnammayel** [comment]
okay sounds good! let's take note then and create an issue to improve the layout in Q1. Thanks
